codelighthouse/CodeLighthouseWebHandler.py

```diff
@@ -19,6 +19,25 @@ def send_error(self, *args, **kwargs):
 
         url = f"{self.BASE_URL}/{self.version}/error"
 
+        # ENSURE THAT THE USER DATA IS SERIALIZABLE TO JSON
+        # TRY TO JSON SERIALIZE THE ENTIRE REQUEST. IF IT FAILS, IT'LL BE BECAUSE OF THE PASSED user_data.
+        # CAN'T JUST TRY TO SERIALIZE THE user_data TO JSON BECAUSE IT COULD BE A NATIVE TYPE OR STRING.
+        try:
+            json.dumps(kwargs)
+        # CATCH ERROR THROWN IF OBJECT IS NOT SERIALIZABLE
+        except TypeError as e:
+
+            # CHECK IF IT HAS AN __DICT__ ATTRIBUTE AND IF SO USE THAT
+            if hasattr(kwargs['user_data'], '__dict__'):
+                kwargs['user_data'] = kwargs['user_data'].__dict__
+            else:
+                kwargs['user_data'] = {'error': 'The passed data was not JSON-serializable'}
+                print('CODELIGHTHOUSE: additional data specified in CodeLighthouse.error() could not be serialized'
+                      'to JSON. Omitting it from the report.')
+
+
```

Comment:
You'll need another `json.dumps` in between lines 32 and 33

Reply:
55877a5cf3d1705b82e81f3c778cfe67249bfca4 @willtraweek